src/hooks/useObserver.ts

```diff
@@ -0,0 +1,24 @@
+import { RefObject, useEffect, useRef } from 'react';
+
+export const useObserver = (observerTarget: RefObject<any>, canLoad: boolean, loading: boolean, callback: () => void) => {
+  const observer = useRef<IntersectionObserver>();
+
+  useEffect(() => {
+    if (loading) return;
+
+    const cb = (entries: IntersectionObserverEntry[]) => {
+      if (entries[0].isIntersecting && canLoad) {
+        callback();
+      }
+    }
+
+    observer.current = new IntersectionObserver(cb);
+    observer.current.observe(observerTarget.current);
+
+    return () => {
+      if (observer.current) {
+        observer.current.disconnect();
+      }
+    };
```

Comment:
You don't need to check for observer existence, because of the early return on line 7.
```suggestion
    return observer.current.disconnect;
```

Reply:
I wasn't able to use the suggested version as is because it failed for me with the "Illegal invocation" runtime error, but I removed the surrounding if-clause.